src/main/java/org/sagebionetworks/bridge/udd/synapse/SynapseDownloadFromTableTask.java

```diff
@@ -0,0 +1,421 @@
+    private static final String QUERY_TEMPLATE =
+            "SELECT * FROM %s WHERE healthCode = '%s' AND uploadDate >= '%s' AND uploadDate <= '%s'";
+
+    // Task parameters. Params is passed in by constructor. Context is created by this task.
+    private final SynapseDownloadFromTableParameters params;
+    private final SynapseDownloadFromTableContext ctx = new SynapseDownloadFromTableContext();
+
+    // Helpers and config objects. Originates from Spring configs and is passed in through setters using a similar
+    // pattern.
+    private FileHelper fileHelper;
+    private SynapseHelper synapseHelper;
+
+    /**
+     * Constructs this task with the specified task parameters
+     *
+     * @param params
+     *         task parameters
+     */
+    public SynapseDownloadFromTableTask(SynapseDownloadFromTableParameters params) {
+        this.params = params;
+    }
+
+    /**
+     * Wrapper class around the file system. Used by unit tests to test the functionality without hitting the real file
+     * system.
+     */
+    public final void setFileHelper(FileHelper fileHelper) {
+        this.fileHelper = fileHelper;
+    }
+
+    /** Synapse helper, used to download CSV and bulk file download from Synapse. */
+    public final void setSynapseHelper(SynapseHelper synapseHelper) {
+        this.synapseHelper = synapseHelper;
+    }
+
+    /**
+     * Executes the SynapseDownloadFromTableTask. Returns the list of files downloaded. These files all live in the
+     * temp directory passed in from the task parameters.
+     *
+     * @return list of files downloaded
+     */
+    @Override
+    public SynapseDownloadFromTableResult call() {
+        try {
+            downloadCsv();
+            if (filterNoDataCsvFiles()) {
+                // return an empty result, to signify no data
+                return new SynapseDownloadFromTableResult.Builder().build();
+            }
+            getColumnInfoFromCsv();
+
+            if (ctx.getColumnInfo().getFileHandleColumnIndexSet().isEmpty()) {
+                LOG.info("No file handles columns in file " + ctx.getCsvFilePath() +
+                        ". Skipping extracting and downloading file handles.");
+            } else {
+                extractFileHandleIdsFromCsv();
+
+                if (ctx.getFileHandleIdSet().isEmpty()) {
+                    // This is rare but possible.
+                    LOG.info("No file handles to download for file " + ctx.getCsvFilePath() +
+                            ". Skipping downloading file handles.");
+                } else {
+                    bulkDownloadFileHandles();
+                }
+            }
+
+            editCsv();
+
+            return new SynapseDownloadFromTableResult.Builder().withCsvFile(ctx.getCsvFile())
+                    .withBulkDownloadFile(ctx.getBulkDownloadFile()).build();
+        } catch (RuntimeException ex) {
```

Comment:
We only want to clean up the files for an exception. In the normal case, the files are passed back to the caller, and the caller is in charge of deleting them. So we still need a catch clause. That said, I can simplify this to re-throwing the exception instead of mindlessly wrapping it in a new exception.

We still want to catch RuntimeException though, so even if something unexpected happens, we run cleanup.

(I also considered not cleaning up at all, so we can check the files and see why they failed. I decided against it, because this might clog up the file system if it happens to frequently.)